wpilibj/src/main/java/edu/wpi/first/wpilibj/XboxController.java

```diff
@@ -93,170 +108,136 @@ public boolean getBumper(Hand hand) {
   }
 
   /**
-   * This is not supported for the XboxController. This method is only here to complete the
-   * GenericHID interface.
-   *
-   * @param hand This parameter is ignored for the Joystick class and is only here to complete the
-   *             GenericHID interface.
-   * @return The state of the trigger (always false)
-   */
-  @SuppressWarnings("PMD.UnusedFormalParameter")
-  public boolean getTrigger(Hand hand) {
-    return false;
-  }
-
-  /**
-   * This is not supported for the XboxController. This method is only here to complete the
-   * GenericHID interface.
-   *
-   * @param hand This parameter is ignored for the Joystick class and is only here to complete the
-   *             GenericHID interface.
-   * @return The state of the top button (always false)
-   */
-  @SuppressWarnings("PMD.UnusedFormalParameter")
-  public boolean getTop(Hand hand) {
-    return false;
-  }
-
-  /**
-   * Get the button value (starting at button 1).
-   *
-   * <p>The appropriate button is returned as a boolean value.
-   *
-   * @param button The button number to be read (starting at 1).
-   * @return The state of the button.
-   */
-  public boolean getRawButton(final int button) {
-    return m_ds.getStickButton(getPort(), (byte) button);
-  }
-
-  /**
-   * Get the trigger axis value of the controller.
+   * Read the value of the stick button on the controller.
    *
+   * @deprecated Use getButton() instead.
    * @param hand Side of controller whose value should be returned.
-   * @return The trigger axis value of the controller.
+   * @return The state of the button.
    */
-  public double getTriggerAxis(Hand hand) {
+  @Deprecated
+  public boolean getStickButton(Hand hand) {
     if (hand.equals(Hand.kLeft)) {
-      return getRawAxis(2);
+      return getRawButton(9);
     } else {
-      return getRawAxis(3);
+      return getRawButton(10);
```

Comment:
Do we want to replace these methods with something that calls the new API?

Reply:
Normally, I would have. However, calling getButton(Button.kStickButtonLeft) in this case calls getButtonIndex() internally. Since we already know which button it is, we can avoid the call to getButtonIndex().